### PR TITLE
Follow INLINE pragmas when adding loopbreakers

### PR DESCRIPTION
**Pull request by TheMatten**
This PR adds function-specific annotations for enabling/disabling loopbreakers and corresponding top level flags for changing default behaviour, plus some small fixes.

### CODE REVIEW

**Comment by TheMatten** [comment]
(I plan to add documentation for whole library as a separate PR)

File: src/Loopbreaker/InlineRecCalls.hs [hunk 1]
```diff
@@ -1,38 +1,69 @@
...
+-- | Forces compiler to inline functions by creating loopbreaker with NOINLINE
+-- pragma, changing recursive calls to use it and by adding INLINE pragma to
+-- the original function.
+action :: MonadInline m
+       => [CommandLineOption] -> HsGroup GhcRn -> m (HsGroup GhcRn)
+action opts group = do
+  -- Last option takes precedence, on by default
+  let isDefaultOn = fromMaybe True $ getLast
+                  $ flip foldMap opts $ Last . \case
+                      "default_on"  -> Just True
```
Comment (isovector) [inline src/Loopbreaker/InlineRecCalls.hs]: I think these could use stronger names. Like `break-all-loops` or something.

File: src/Loopbreaker/InlineRecCalls.hs [hunk 2]
```diff
@@ -1,38 +1,69 @@
...
+                      "default_off" -> Just False
+                      _             -> Nothing
+
   dyn_flags <- getDynFlags
 
-  bool pure inlineRecCalls (optLevel dyn_flags > 0) group
+  if optLevel dyn_flags > 0
+    then do
+      liftIO $ showPass dyn_flags "Add loopbreakers"
```
Comment (isovector) [inline src/Loopbreaker/InlineRecCalls.hs]: Possibly `"Break Loops"`?

**Review by isovector — APPROVED** [review]
Looks great! I'd love some more inspection tests that show the loopbreaking actually helps optimize things away.

File: src/Loopbreaker/InlineRecCalls.hs (hunk 1 above)
Comment (TheMatten) [reply src/Loopbreaker/InlineRecCalls.hs]: Ok.

File: src/Loopbreaker/InlineRecCalls.hs (hunk 2 above)
Comment (TheMatten) [reply src/Loopbreaker/InlineRecCalls.hs]: Probably better :slightly_smiling_face:

**Comment by TheMatten** [comment]
I'm wondering what should I use as an example - functions from `polysemy` would depend on it's machinery...

**Comment by isovector** [comment]
Yeah, I don't have a good example. If you want to stress test it in the wild, you could add `loopbreaker` to `polysemy` and use it to break all of the loops by hand. If the tests _there_ still pass, that's a pretty good sign!

**Comment by isovector** [comment]
That's the only way I found out that the coreplugin version didn't help the inlining process!

**Comment by TheMatten** [comment]
I'm going to add support for 8.4.4 in this PR too to make things easier.

**Comment by TheMatten** [comment]
I forgot (again) that `renamedResultAction` was introduced in 8.6....
So no `loopbreakers` for 8.4.

**Comment by TheMatten** [comment]
@isovector do you think it's ready?

**Comment by isovector** [comment]
:shipit:

**Pull request merged**